src/Reception.jsx

```diff
@@ -0,0 +1,336 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import { Link } from 'react-router-dom';
+import moment from 'moment';
+import Header from './Header';
+import Table from './Table';
+import ModalDelete from './ModalDelete';
+import './Reception.scss';
+import ModalChange from './ModalChange';
+import _ from 'lodash';
+
+const Reception = () => {
+  const [allAppoint, setAllAppoint] = useState([]);
+  const [activeFilter, setActiveFilter] = useState(false);
+  const [activeDelete, setActiveDelete] = useState(false);
+  const [activeChange, setActiveChange] = useState(false);
+  const [idDelete, setIdDelete] = useState('');
+  const [sortCheck, setSortCheck] = useState(false);
+  const [direction, setDirection] = useState(false);
+  const [idChange, setIdChange] = useState('');
+  const [itemApp, setItemApp] = useState('');
+  const [dateFrom, setDateFrom] = useState('');
+  const [dateBy, setDateBy] = useState('');  
+  const [nameChange, setNameChange] = useState('');
+  const [doctorChange, setDoctorChange] = useState('');
+  const [dateChange, setDateChange] = useState('');
+  const [complaintsChange, setComplaintsChange] = useState('itemApp.complaints');
+
+  const url = 'http://localhost:8000';
+
+  const deleteFunction = async (id) => {
+    await axios.delete(`${url}/deleteAppoint?_id=${id}`).then((res) => {
+      const deleteAppoint = allAppoint.filter(
+        (allAppoint) => allAppoint._id !== id
+      );
+      setIdDelete('');
+      setAllAppoint(deleteAppoint);
+    });
+  };
+
+  useEffect(async () => {
+    await axios.get(`${url}/allAppoints`).then((res) => {
+      setAllAppoint(res.data);
+    });
+  }, []);
+
+  const appointment = async (e) => {
+    e.preventDefault();
+    const data = new FormData(e.target);
+    const name = data.get('name');
+    const doctor = data.get('doctor');
+    const date = data.get('date');
+    const complaints = data.get('complaints');
+    if (name && doctor && date && complaints) {
+      const reception = [...allAppoint];
+      const newRec = {
+        name: name,
+        doctor: doctor,
+        date: date,
+        complaints: complaints,
+      };
+      await axios.post(`${url}/createAppoint`, newRec).then((res) => {
+        newRec._id = res.data._id;
+        reception.push(newRec);
+        setAllAppoint(reception);
+      });
+      e.target.reset();
+    } else alert('Введите все поля');
+  };
+
+  const sortingReception = (e) => {
```

Comment:
Firstly, sorting should be set immediately when the page is updated. And by default it is ascending. 
Secondly, it must be universal. Then the code will be reduced by 3 times. And if a new field is added to the table, it will not be necessary to add sorting for it either. 
To do this, you can add type for each field of the table. To do this, you can add a type for each field of the table. And pass the type to "sortingReception"